### PR TITLE
Adds types for the 'node-postal' package

### PR DESCRIPTION
**Pull request by joaocarmo**
Adds types for the [node-postal](https://github.com/openvenues/node-postal) package.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/node-postal/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export const parser: {
+    parse_address: typeof parse_address;
+};
+
+export const postal: {
+    expand: typeof expand;
+    parser: typeof parser;
+};
+
+export as namespace postal;
```
Comment (peterblazejewicz) [inline types/node-postal/index.d.ts:51]:
```suggestion

export as namespace postal;
```
this is when module can be used in global scripts via symbol export, say within browser environment,
not really applicable here I think

File: types/node-postal/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+export const parser: {
+    parse_address: typeof parse_address;
+};
+
+export const postal: {
+    expand: typeof expand;
+    parser: typeof parser;
+};
```
Comment (peterblazejewicz) [inline types/node-postal/index.d.ts:49]: those are internal symbols, the module exports only two symbols, nothing more IMO:
```suggestion
declare function expand_address(address: string): string[];

declare function parse_address(address: string): PostalResult[];

export const expand: {
    expand_address: typeof expand_address;
};

export const parser: {
    parse_address: typeof parse_address;
};

export {};
```
https://github.com/openvenues/node-postal/blob/master/index.js#L3-L4

File: types/node-postal/index.d.ts (hunk 1 above)
Comment (joaocarmo) [reply types/node-postal/index.d.ts]: Ah! That makes sense. Thank you for pointing that out!

File: types/node-postal/index.d.ts (hunk 2 above)
Comment (joaocarmo) [reply types/node-postal/index.d.ts]: That is correct, no need to export extra stuff. Thank you for the `export {}` suggestion!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@joaocarmo thx!

**Comment by joaocarmo** [comment]
Ready to merge

**Pull request merged**